src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -264,92 +265,42 @@ public void paintBorder(Component c, Graphics g, int x, int y,
                 shadow = MetalLookAndFeel.getControlInfo();
             }
 
-            AffineTransform at = null;
-            Stroke oldStk = null;
-            boolean resetTransform = false;
-            int stkWidth = 1;
-            double scaleFactor = 1;
-
-            if (g instanceof Graphics2D g2d) {
-                at = g2d.getTransform();
-                scaleFactor = at.getScaleX();
-                oldStk = g2d.getStroke();
-
-                // if m01 or m10 is non-zero, then there is a rotation or shear
-                // skip resetting the transform
-                resetTransform = ((at.getShearX() == 0) && (at.getShearY() == 0));
-
-                if (resetTransform) {
-                    g2d.setTransform(new AffineTransform());
-                    stkWidth = clipRound(Math.min(at.getScaleX(), at.getScaleY()));
-                    g2d.setStroke(new BasicStroke((float) stkWidth));
-                }
-            }
-
-            int xtranslation;
-            int ytranslation;
-            int width;
-            int height;
-
-            if (resetTransform) {
-                double xx = at.getScaleX() * x + at.getTranslateX();
-                double yy = at.getScaleY() * y + at.getTranslateY();
-                xtranslation = clipRound(xx);
-                ytranslation = clipRound(yy);
-                width = clipRound(at.getScaleX() * w + xx) - xtranslation;
-                height = clipRound(at.getScaleY() * h + yy) - ytranslation;
-            } else {
-                xtranslation = x;
-                ytranslation = y;
-                width = w;
-                height = h;
-            }
-            g.translate(xtranslation, ytranslation);
-
             // scaled border
-            int thickness = (int) Math.ceil(4 * scaleFactor);
+            int thickness = (int) Math.ceil(4 * scale);
 
             g.setColor(background);
             // Draw the bulk of the border
             for (int i = 0; i <= thickness; i++) {
-                g.drawRect(i, i, width - (i * 2), height - (i * 2));
+                g.drawRect(i, i, w - (i * 2), h - (i * 2));
```

Comment:
The keep the old code intact, use `width` and `height` as parameter names in `paintUnscaledBorder` method.

The same is true for `scaleFactor` above.